npm/webpack-dev-server/test/makeWebpackConfig.spec.ts

```diff
@@ -38,4 +43,47 @@ describe('makeWebpackConfig', () => {
     expect(actual.output.publicPath).to.eq('/test-public-path/')
     snapshot(actual)
   })
+
+  it('ignores userland webpack `output.publicPath` and `devServer.overlay` with webpack-dev-server v4', async () => {
+    const devServerConfig: WebpackDevServerConfig = {
+      specs: [],
+      cypressConfig: {
+        isTextTerminal: false,
+        projectRoot: '.',
+        supportFile: '/support.js',
+        devServerPublicPathRoute: '/test-public-path', // This will be overridden by makeWebpackConfig.ts
+      } as Cypress.PluginConfigOptions,
+      webpackConfig: {
+        output: {
+          publicPath: '/this-will-be-ignored',
+        },
+        devServer: {
+          magicHtml: true,
```

Comment:
This was added in Webpack 4.1 only. Will we break user config for Webpack 4? Is there a sys test for Webpack 4.0?

Reply:
@JessicaSachs this is a unit test. I'm just verifying that user defined config will be kept.

Reply:
Yeah I saw that was just like "why **this** flag?". Cool.